Source/WebCore/Modules/indexeddb/IDBKeyData.cpp

```diff
@@ -244,33 +230,33 @@ bool IDBKeyData::decode(KeyedDecoder& decoder, IDBKeyData& result)
         result.m_value = ThreadSafeDataBuffer::create(WTFMove(bytes));
         return true;
     }
+    case IndexedDB::KeyType::Array:
+        auto arrayFunction = [](KeyedDecoder& decoder, IDBKeyData& result) {
+            return decode(decoder, result);
+        };
 
-    ASSERT(result.m_type == IndexedDB::KeyType::Array);
-
-    auto arrayFunction = [](KeyedDecoder& decoder, IDBKeyData& result) {
-        return decode(decoder, result);
-    };
-    
-    result.m_value = Vector<IDBKeyData>();
-    return decoder.decodeObjects("array"_s, std::get<Vector<IDBKeyData>>(result.m_value), arrayFunction);
+        result.m_value = Vector<IDBKeyData>();
+        return decoder.decodeObjects("array"_s, std::get<Vector<IDBKeyData>>(result.m_value), arrayFunction);
+    }
+    RELEASE_ASSERT_NOT_REACHED();
 }
 
 int IDBKeyData::compare(const IDBKeyData& other) const
 {
-    if (m_type == IndexedDB::KeyType::Invalid) {
-        if (other.m_type != IndexedDB::KeyType::Invalid)
+    if (type() == IndexedDB::KeyType::Invalid) {
+        if (other.type() != IndexedDB::KeyType::Invalid)
             return -1;
-        if (other.m_type == IndexedDB::KeyType::Invalid)
+        if (other.type() == IndexedDB::KeyType::Invalid)
             return 0;
-    } else if (other.m_type == IndexedDB::KeyType::Invalid)
+    } else if (other.type() == IndexedDB::KeyType::Invalid)
         return 1;
 
-    // The IDBKey::m_type enum is in reverse sort order.
-    if (m_type != other.m_type)
-        return m_type < other.m_type ? 1 : -1;
+    // The IDBKey::type() enum is in reverse sort order.
+    if (type() != other.type())
+        return type() < other.type() ? 1 : -1;
 
     // The types are the same, so handle actual value comparison.
-    switch (m_type) {
+    switch (type()) {
```

Comment:
Should probably use a local variable type to avoid invoking `type()` multiple times.